### DISCUSSION TITLE
Spatie Media Library Plugin and custom properties

### DISCUSSION THREAD

**Discussion by Fot0n** [filamentphp/filament#4714] [Help]
I want to add descriptions to the uploaded files in the form and save them to the custom properties of Spatie Medialibrary. The plugin has the ability to save hard-coded properties:
```
use Filament\Forms\Components\SpatieMediaLibraryFileUpload;
SpatieMediaLibraryFileUpload::make('attachments')
    ->multiple()
    ->customProperties(['zip_filename_prefix' => 'folder/subfolder/']),
```
Please tell me how I can set properties from the form.

**Comment by jeverington** [comment]
I raised a similar question [here](https://github.com/filamentphp/filament/discussions/4533). Not sure of the answer, but it would be really useful to be able to do.

**Comment by digitall-it** [comment]
I am here for the same reason: I would like to attach some fields to each media upload and have the ability to manipulate the input to create custom properties. For example, incorporate somehow [Image Focal Point Picker](https://filamentphp.com/plugins/focal-point-picker) or choose the file name, add tags to images and so on.

**Reply by svendehens** [reply]
Hi, I know this has been a while. But I'm here with the same question. Did you ever find a good solution? I'd rather use the Spatie Media Library than switch to using the Curator of Media Library plugins. Thanks! Sven

**Reply by digitall-it** [reply]
No, couldn’t find a way to perform that, I’m in your situation too still at this time.

**Comment by ArtMin96** [comment]
Maybe you can call `$this->form->getState()`?

**Comment by madushancs** [comment]
When edit this will work,

`->customProperties(['team_id' => $form->model->description])`

**Comment by annya-a** [comment]
Try this one:
```
use Filament\Forms\Get;
SpatieMediaLibraryFileUpload::make('attachments')
                    ->collection('image')
                    ->customProperties(function (Get $get) {
                        return ['credits' => $get('description')];
                    }),

**Reply by digitall-it** [reply]
This wouldn't work with multiple images, but seems to be a solid approach when there's one.
